pkg/controller/applicationintegration/sqs/controller.go

```diff
@@ -0,0 +1,244 @@
+	if err := c.kube.Get(ctx, n, s); err != nil {
+		return nil, errors.Wrap(err, errGetProviderSecret)
+	}
+
+	queueClient, err := c.newClientFn(ctx, s.Data[p.Spec.CredentialsSecretRef.Key], p.Spec.Region, awsclients.UseProviderSecret)
+	return &external{client: queueClient, kube: c.kube}, errors.Wrap(err, errQueueClient)
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mg.(*v1alpha1.Queue)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotQueue)
+	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{}, nil
+	}
+
+	resAttributes, err := e.client.GetQueueAttributesRequest(&awssqs.GetQueueAttributesInput{
+		QueueUrl:       aws.String(meta.GetExternalName(cr)),
+		AttributeNames: []awssqs.QueueAttributeName{awssqs.QueueAttributeNameAll},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(sqs.IsErrorNotFound, err), errGetQueueAttributesFailed)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Available())
+
+	cr.Status.AtProvider = v1alpha1.QueueObservation{
+		ARN: resAttributes.Attributes[v1alpha1.AttributeQueueArn],
+	}
+
+	resTags, err := e.client.ListQueueTagsRequest(&awssqs.ListQueueTagsInput{
+		QueueUrl: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errListQueueTagsFailed)
+	}
+
+	sqs.LateInitialize(&cr.Spec.ForProvider, resAttributes.Attributes, resTags.Tags)
+	current := cr.Spec.ForProvider.DeepCopy()
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: sqs.IsUpToDate(cr.Spec.ForProvider, resAttributes.Attributes, resTags.Tags),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.Queue)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotQueue)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Creating())
+
+	// FIFO queues names should end with ".fifo"
+	if aws.BoolValue(cr.Spec.ForProvider.FifoQueue) && !strings.HasSuffix(cr.Spec.ForProvider.Name, fifoQueueSuffix) {
+		return managed.ExternalCreation{}, errors.New(errInvalidNameForFifoQueue)
+	}
+
+	createResp, err := e.client.CreateQueueRequest(&awssqs.CreateQueueInput{
+		Attributes: sqs.GenerateCreateAttributes(&cr.Spec.ForProvider),
+		QueueName:  aws.String(cr.Spec.ForProvider.Name),
+		Tags:       sqs.GenerateQueueTags(cr.Spec.ForProvider.Tags),
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateFailed)
+	}
+
+	meta.SetExternalName(cr, aws.StringValue(createResp.QueueUrl))
```

Comment:
I see that the queue URL is the unique identifier for a queue, so we need to use that when observing, updating, and deleting the queue. The docs for the below API imply that it's possible to look up a queue URL using only a queue name. Could we store just the queue name in our external name annotation, then use the API below to lookup the URL based on that name when observing, updating, or deleting? This would let us get rid of the `spec.forProvider.name` field and just use the external name annotation, which would be ideal.

https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_GetQueueUrl.html

Reply:
I have made the required changes after setting the external-name to the queue name and removing the `Name` field from `ForProvider`